src/devices/schraeder.c

```diff
@@ -252,6 +252,95 @@ static int schrader_SMD3MA4_decode(r_device *decoder, bitbuffer_t *bitbuffer)
     return 1;
 }
 
+/**
+TPMS Model: Schrader Electronics MRXBC5A4 and MRXBMW433TX1
+Contributed by: Ilias Daradimos.
+
+Probable packet payload:
+
+    W SSSSSSSSSSSSS S FFF IIIIIIIIIIIIIIIIIIIIIIII PPPPPPPP ?? TTTTTTTT
+
+- W: 1 bit wake
+- S: 13 sync bit, 1 start bit
+- F: 3 bits, might contain status and battery flags.
+- I: id (24 bits)
+- P: pressure 8 bits 2kPa/bit
+- C: Checksum
+- T: 8 bits temperature offset by 50, degrees C -50 to 205
+*/
+static int schrader_MRXBC5A4_decode(r_device *decoder, bitbuffer_t *bitbuffer)
+{
+    data_t *data;
+    uint8_t b[7];
+    unsigned int serial_id;
+    char id_str[19];
+    uint8_t flags;
+    char flags_str[9];
+    unsigned int pressure;    // mbar
+    int temperature; // degree C
+    uint8_t checksum, bits, n;
+    uint8_t parity = 0;
+
+    /* Check for incorrect number of bits received */
+    if (bitbuffer->bits_per_row[0] != 61) {
+      if (decoder->verbose > 1) {
+        fprintf(stderr, "%s: length error %d \n", __func__, bitbuffer->bits_per_row[0]);
+      }
+      return DECODE_ABORT_LENGTH;
+    }
+    /* Discard the first 15 bits */
+    bitbuffer_extract_bytes(bitbuffer, 0, 16, b, 46);
+    // check serial value not zero
+    if ( !b[0] && !b[1] && !b[2] && !b[3] ) {
+        if (decoder->verbose > 1) {
+            fprintf(stderr, "%s: DECODE_FAIL_SANITY data all 0x00\n", __func__);
+        }
+        return DECODE_FAIL_SANITY;
+    }
+
+    // Check parity
+    for (n=0; n<=4; n++){
+      bits = b[n] & 0xAA;
+      if (n == 4) bits &= 0xf8;
+      while (bits)
+      {
```

Comment:
Usual bit count is repeated folding (no double loop):
```
parity = b[0] ^ b[1] ^ b[2] ^ b[3] ^ (b[4] & 0xf8);
parity ^= parity >> 4; // fold
parity ^= parity >> 2;
// odd_bits_parity = parity & 1; //
parity = parity & 2; // even bits only
```

Reply:
The parity that validates is on the odd bits so parity bit can be included in calculation producing a 0 when parity is valid and code can look like:
```c
    parity = b[0] ^ b[1] ^ b[2] ^ b[3] ^ (b[4] & 0xfe);
    parity ^= parity >> 4; // fold
    parity ^= parity >> 2;
    parity = parity & 1; // Odd bits

    if (parity) {
      if (decoder->verbose > 1) {
          fprintf(stderr, "%s: Parity fail\n", __func__);
      }
      return DECODE_FAIL_MIC;
    }
`